Add more props and formatting to the shared summary component for report pages

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

Parts of #242, #244. Some adjustments of shared component **MetricNumber** used in both report pages.

- Forward **MetricNumber** props of `href`, `selected`, and `prevValue` to **SummaryNumber**
- Format `value` and `prevValue` within the **MetricNumber**
- Change selected metric and display selected style when clicking on **SummarySection** metrics for the Products Report page

### Screenshots:

![Kapture 2021-05-07 at 17 20 09](https://user-images.githubusercontent.com/17420811/117428282-821f3c00-af58-11eb-8136-20de85646dc7.gif)

### Detailed test instructions:

1. Head to the Products Report page `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Freports%2Fproducts`
2. The currency format displayed on the summary section should align with store's currency setting
3. The default selected summary metric should be the first metric
4. Click on summary metrics, the selected metric and table's order column should be changed respectively. And vice versa, clicking on product table headers should change the selected metric as well

eason9487 committed:
Forward MetricNumber props of href, selected, and prevValue to SummaryNumber

eason9487 committed:
Make currency relevant hooks return stable references

eason9487 committed:
Format value and prevValue within MetricNumber

eason9487 committed:
Change selected metric and display selected style when clicking on SummarySection metrics

eason9487 committed:
Fix lint error

tomalec reviewed (approved):
Nice :)
Tested locally, reviewed the code, works and LGTM.

eason9487 committed:
Change metric structure from array to disc object

Addressed comments:

- https://github.com/woocommerce/google-listings-and-ads/pull/565#discussion_r628119908
- https://github.com/woocommerce/google-listings-and-ads/pull/565#discussion_r628123371
